macos/FluentUI/AvatarView/AvatarView.swift

```diff
@@ -390,6 +413,20 @@ open class AvatarView: NSView {
 		}
 	}
 
+	open override func viewWillMove(toWindow newWindow: NSWindow?) {
+		super.viewWillMove(toWindow: newWindow)
+
+		guard let window = newWindow else {
+			appearanceObserver = nil
+			return
+		}
+		appearanceObserver = window.observe(\.effectiveAppearance) { [weak self] (window, _) in
+			guard let strongSelf = self else {
+				return
+			}
+			strongSelf.updateAppearance(window.appearance ?? NSAppearance.current)
```

Comment:
**window.effectiveAppearance** should do the trick for you here:

`strongSelf.updateAppearance(window.effectiveAppearance)`

Reply:
This worked, thanks. Saved a lot of complexity for testing